In [1]:
from ultralytics import YOLO

In [2]:
model = YOLO("best-detect.pt")

In [3]:
model.benchmark(
    data="data.yaml", # TODO: refactor this ( use one instance of dataset instead copy it )
    # format="onnx"
)
print("Model Benchmark...")

Setup complete  (8 CPUs, 31.5 GB RAM, 114.3/475.3 GB disk)

Benchmarks complete for best-detect.pt on data.yaml at imgsz=640 (342.48s)
Benchmarks legend:  -  Success  -  Export passed but validation failed  -  Export failed
+-------------------------------------------------------------------------------------------------+
|      Format         Status   Size (MB)   metrics/mAP50-95(B)   Inference time (ms/im)   FPS   |
+=================================================================================================+
| 1    PyTorch                 5.2         0.9351                54.5                     18.35 |
| 2    TorchScript             10.4        0.9281                247.42                   4.04  |
| 3    ONNX                    10.1        0.9281                55.05                    18.16 |
| 4    OpenVINO                10.3        0.9281                208.15                   4.8   |
| 5    TensorRT                0.0         -                     -                    

Setup complete  (8 CPUs, 31.5 GB RAM, 114.3/475.3 GB disk)

- Benchmarks complete for best-detect.pt on data.yaml at imgsz=640 (342.48s)
- Benchmarks legend:  -  Success  -  Export passed but validation failed  -  Export failed

| Format | Status                                                                                   | Size (MB) | metrics/mAP50-95(B) | Inference time (ms/im) | FPS   |
|--------|------------------------------------------------------------------------------------------|-----------|---------------------|------------------------|-------|
| 1      | PyTorch                                                                                  | 5.2       | 0.9351              | 54.5                   | 18.35 |
| 2      | TorchScript                                                                              | 10.4      | 0.9281              | 247.42                 | 4.04  |
| 3      | ONNX                                                                                     | 10.1      | 0.9281              | 55.05                  | 18.16 |
| 4      | OpenVINO                                                                                 | 10.3      | 0.9281              | 208.15                 | 4.8   |
| 5      | TensorRT                                                                                 | 0.0       | -                   | -                      | -     |
| 6      | CoreML                                                                                   | 0.0       | -                   | -                      | -     |
| 7      | PaddlePaddle                                                                             | 0.0       | -                   | -                      | -     |
| 8      | MNN                                                                                      | 0.0       | -                   | -                      | -     |
| 9      | NCNN                                                                                     | 10.0      | 0.9281              | 308.87                 | 3.24  |
| 10     | IMX                                                                                      | 0.0       | -                   | -                      | -     |
| 11     | RKNN                                                                                     | 0.0       | -                   | -                      | -     |
| 12     | ExecuTorch                                                                               | 10.1      | -                   | -                      | -     |
| 13     | Axelera AI                                                                               | 0.0       | -                   | -                      | -     |
| 14     | DEEPX                                                                                    | 0.0       | -                   | -                      | -     |
| 15     | Qualcomm QNN                                                                             | 0.0       | -                   | -                      | -     |
| 16     | LiteRT                                                                                   | 0.0       | -                   | -                      | -     |
| 17     | Hailo                                                                                    | 0.0       | -                   | -                      | -     |

Model Benchmark...